File: src/test/java/StopWordsTest.java
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class StopWordsTest {

    private StopWords testSubject;

    @BeforeEach
    public void setUp() {
        testSubject = new StopWords("test-stopwords.txt");
    }

    @Test
    void testInit_givenFileName_thenAllDefinedWordsArePresentInSetOfStopWords() {
        assertEquals(4, testSubject.getAllStopWords().size());
        assertTrue(testSubject.getAllStopWords().contains("the"));
        assertTrue(testSubject.getAllStopWords().contains("a"));
        assertTrue(testSubject.getAllStopWords().contains("on"));
        assertTrue(testSubject.getAllStopWords().contains("off"));
    }

    @Test
    void testIsStopWord_givenStopWord_thenReturnTrue() {
        assertTrue(testSubject.isStopWord("the"));
        assertTrue(testSubject.isStopWord("The"));
        assertTrue(testSubject.isStopWord("THE"));
        assertTrue(testSubject.isStopWord("a"));
        assertTrue(testSubject.isStopWord("on"));
        assertTrue(testSubject.isStopWord("off"));
    }

    @Test
    void testIsStopWord_givenNotStopWord_thenReturnFalse() {
        assertFalse(testSubject.isStopWord(null));
        assertFalse(testSubject.isStopWord(""));
        assertFalse(testSubject.isStopWord("    "));
        assertFalse(testSubject.isStopWord("something"));
        assertFalse(testSubject.isStopWord("123"));
        assertFalse(testSubject.isStopWord("???"));
    }
}

File: src/test/kotlin/wordcount/StopwordsTest.kt
package wordcount

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Test
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource

class StopwordsTest {

    @ParameterizedTest
    @CsvSource(
        "mary, true",
        "MARy, true",
        " the , true",
        "Hello, false",
        "HelloMARY, false",
        " , false",
    )
    fun `isStopword should mark word correctly`(word: String?, expected: Boolean) {
        val words = listOf(
            "Mary",
            " the",
            "a "
        )

        val stopwords = Stopwords(words)
        val res = stopwords.isStopword(word)

        assertEquals(expected, res)
    }

    @Test
    fun `should consider all words if a file can not be read`() {
        val stopwords = Stopwords.fromResource("unknown-file")

        assertFalse(stopwords.isStopword("a"))
        assertFalse(stopwords.isStopword("the"))
    }
}